common_migration/find_common_migrations.py

```diff
@@ -0,0 +1,444 @@
+#!/usr/bin/env python3
+
+"""
+Find a common migration node between two versions of Spotlight code.
+
+"""
+
+import argparse
+import ast
+import hashlib
+import os
+import re
+import sys
+from typing import Dict, Iterator, List, Optional, Set, Tuple, cast
+
+MIGRATION_NAME_RE = re.compile(r'^\d+_.')
+
+
+class NotAMigrationError(IOError):
+    pass
+
+
+class MigrationNode:
+    """
+    A migration node, with the following attributes.
+
+    hash:         A hash describing the entire subgraph up from the node.
+    app_name:     The app name of the migration.
+    name:         The name of the migration.
+    dependencies: The dependencies for this migration node.
+    dependents:   Nodes that depend on this node.
+    """
+
+    migration_number: int
+
+    def __init__(
+        self,
+        *,
+        hash: bytes,
+        app_name: str,
+        name: str,
+        dependencies: 'List[MigrationNode]',
+        dependents: 'List[MigrationNode]',
+    ):
+        self.hash = hash
+        self.app_name = app_name
+        self.name = name
+        self.dependencies = dependencies
+        self.dependents = dependents
+
+        split_name = name.split('_', 1)
```

Comment:
Not sure you can guarantee this (we already have a migration with a space in it). Suggest using '^\d+' or similar instead